fvm/src/kernel/default.rs

```diff
@@ -304,32 +304,25 @@ impl<C> MessageOps for DefaultKernel<C>
 where
     C: CallManager,
 {
-    fn msg_caller(&self) -> ActorID {
-        self.caller
-    }
-
-    fn msg_origin(&self) -> ActorID {
-        self.call_manager.origin()
-    }
-
-    fn msg_receiver(&self) -> ActorID {
-        self.actor_id
-    }
-
-    fn msg_method_number(&self) -> MethodNum {
-        self.method
-    }
-
-    fn msg_value_received(&self) -> TokenAmount {
-        self.value_received.clone()
-    }
-
-    fn msg_gas_premium(&self) -> TokenAmount {
-        self.call_manager.gas_tracker().gas_premium()
-    }
-
-    fn msg_gas_limit(&self) -> u64 {
-        self.call_manager.gas_tracker().gas_limit().round_down() as u64
+    fn msg_context(&self) -> Result<MessageContext> {
+        // TODO: charge gas.
+        Ok(MessageContext {
```

Comment:
We could consider creating this structure once on initialization, but that may not be worth it.

Reply:
Depends on how much it costs!

Reply:
We could cache in first access and reuse and reduce gas charge for subsequent accesses.

Reply:
Yep, you're right. Profile first.